avm/res/hybrid-compute/machine/main.bicep

```
@@ -0,0 +1,256 @@
+metadata name = 'Hybrid Compute Machines'
+metadata description = 'This module deploys an Arc Machine for use with Arc Resource Bridge for Azure Stack HCI or VMware. In these scenarios, this resource module will be used in combination with another resource module to create the require Virtual Machine Instance extension resource on this Arc Machine resource. This module should not be used for other Arc-enabled server scenarios, where the Arc Machine resource is created automatically by the onboarding process.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the Arc machine to be created. You should use a unique prefix to reduce name collisions in Active Directory.')
+param name string
+
+@description('Required. Kind of Arc machine to be created. Possible values are: HCI, SCVMM, VMware.')
+param kind string
+
+@description('Conditional. The resource ID of an Arc Private Link Scope which which to associate this machine. Required if you are using Private Link for Arc and your Arc Machine will resolve a Private Endpoint for connectivity to Azure.')
+param privateLinkScopeResourceId string = ''
+
+@description('Optional. Parent cluster resource ID (Azure Stack HCI).')
+param parentClusterResourceId string = ''
+
+@description('Optional. The GUID of the on-premises virtual machine from your hypervisor.')
+param vmId string = ''
+
+@description('Optional. The Public Key that the client provides to be used during initial resource onboarding.')
+param clientPublicKey string = ''
+
+@description('Optional. VM guest patching orchestration mode. \'AutomaticByOS\' & \'Manual\' are for Windows only, \'ImageDefault\' for Linux only.')
+@allowed([
+  'AutomaticByPlatform'
+  'AutomaticByOS'
+  'Manual'
+  'ImageDefault'
+  ''
+])
+param patchMode string = ''
```

Instead of an empty default, did you try the following? 

```suggestion
@description('Optional. VM guest patching orchestration mode. \'AutomaticByOS\' & \'Manual\' are for Windows only, \'ImageDefault\' for Linux only.')
@allowed([
  'AutomaticByPlatform'
  'AutomaticByOS'
  'Manual'
  'ImageDefault'
])
param patchMode string?
``` 

If not, Id' suggest it. Saves somes pixels 😉

---

Same comment for the `osType` parameter

---

Switched to the string? approach for both osType and patchMode. I also needed to change the windowsConfiguration and linuxConfiguration conditionals to be case sensitive for this to work--which is ok because @allowed is case-sensitive as well as far as I can tell

---

Hey @mbrat2005,
the main point is the `?`, aka `nullable`. It essentially makes the parameter optional without the need of providing a default.